cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBrokerConfigurationBuilder.java

```diff
@@ -41,10 +43,34 @@
  * generate the configuration file, it is using the PrintWriter.
  */
 public class KafkaBrokerConfigurationBuilder {
+    // Names of environment variables placeholders replaced only in the running container
+    //       => needed both for shared and per-broker configuration
+    private final static String PLACEHOLDER_CERT_STORE_PASSWORD = "${CERTS_STORE_PASSWORD}";
+    private final static String PLACEHOLDER_RACK_ID = "${STRIMZI_RACK_ID}";
+    private final static String PLACEHOLDER_OAUTH_CLIENT_SECRET = "${STRIMZI_%s_OAUTH_CLIENT_SECRET}";
+
+    // Names of environment variables placeholders replaced only in the running container
+    //       => needed only for shared configuration in StatefulSets
+    private final static String PLACEHOLDER_BROKER_ID = "${STRIMZI_BROKER_ID}";
+    private final static String PLACEHOLDER_ADVERTISED_HOSTNAME = "${STRIMZI_%s_ADVERTISED_HOSTNAME}";
+    private final static String PLACEHOLDER_ADVERTISED_PORT = "${STRIMZI_%s_ADVERTISED_PORT}";
+
+    // Additional replaceables are used in the configuration file for node port addresses. These are used for both
+    // per-broker and shared configurations. Listed here for reference only, they are constructed in the
+    // KafkaAssemblyOperator class on the fly based on the user configuration.
+    // * ${STRIMZI_NODEPORT_DEFAULT_ADDRESS}
+    // * ${STRIMZI_NODEPORT_EXTERNALIP_ADDRESS}
+    // * ${STRIMZI_NODEPORT_EXTERNALDNS_ADDRESS}
+    // * ${STRIMZI_NODEPORT_INTERNALIP_ADDRESS}
+    // * ${STRIMZI_NODEPORT_INTERNALDNS_ADDRESS}
+    // * ${STRIMZI_NODEPORT_HOSTNAME_ADDRESS}
+
     private final StringWriter stringWriter = new StringWriter();
     private final PrintWriter writer = new PrintWriter(stringWriter);
     private final Reconciliation reconciliation;
 
+    private String brokerId = PLACEHOLDER_BROKER_ID;
```

Comment:
Why are you assigning the placeholder constant to a variable and using it through the code instead of using the placeholder itself as done with the others?

Reply:
Depending on whether you generate per-broker configuration or shared configuration, the `brokerId` value will be either the placeholder or the actual broker ID. If I wanted to use the placeholder constant directly, the code would be full of something like `if (brokerId == null) use-placeholder else use-actual-id`. This makes it easier -> I just set the `brokerId` either to the placeholder or to the actual broker ID and use the `brokerId` variable everywhere.

Reply:
Yeah I missed that. Thanks.